internal-services/catalog/publish-index-image-task.yaml

```diff
@@ -49,14 +49,53 @@ spec:
         quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
       script: |
         #!/usr/bin/env bash
+        set -euo pipefail
+
         PATH=/bin:/usr/bin:/usr/local/bin
         export PATH
 
-        # do not authenticate if the source is redhat's "registry-proxy" which is unauthenticated.
+        AUTH_PARAM=()
+        # Do not authenticate if the source is Red Hat's "registry-proxy" which is unauthenticated.
         if [[ ! "$(params.sourceIndex)" =~ ^registry-proxy(\-stage)?.engineering.redhat.com ]]; then
             AUTH_PARAM=("--src-creds" "${SOURCE_INDEX_CREDENTIAL}")
         fi
+        TARGET_AUTH="--dest-creds ${TARGET_INDEX_CREDENTIAL}"
+
+        # Check if the target image exists
+        echo "Checking if target image exists: $(params.targetIndex)"
+        set +e
```

Comment:
You can do this without changing the flag, if the command is inside an `if` or using `||` in the end of the command, for instance: 

```
if ! my command; then
  ERROR=1
fi
```
or
```
my command ||  ERROR=$?
```

Reply:
Thanks for suggestion, fixed!